src/java.base/share/classes/java/nio/file/Files.java

```diff
@@ -2803,40 +2803,37 @@ public static Path walkFileTree(Path start,
         try (FileTreeWalker walker = new FileTreeWalker(options, maxDepth)) {
             FileTreeWalker.Event ev = walker.walk(start);
             do {
-                FileVisitResult result;
-                switch (ev.type()) {
-                    case ENTRY :
+                FileVisitResult result = switch (ev.type()) {
+                    case ENTRY -> {
                         IOException ioe = ev.ioeException();
                         if (ioe == null) {
                             assert ev.attributes() != null;
-                            result = visitor.visitFile(ev.file(), ev.attributes());
+                            yield visitor.visitFile(ev.file(), ev.attributes());
                         } else {
-                            result = visitor.visitFileFailed(ev.file(), ioe);
+                            yield visitor.visitFileFailed(ev.file(), ioe);
                         }
-                        break;
-
-                    case START_DIRECTORY :
-                        result = visitor.preVisitDirectory(ev.file(), ev.attributes());
+                    }
+                    case START_DIRECTORY -> {
+                        var r = visitor.preVisitDirectory(ev.file(), ev.attributes());
```

Comment:
Can you all uses of "r" with "res" to make it little bit clear that it's a result, that will get it a bit more consistent with the exist code? Otherwise I think this version is okay.

Reply:
Updated variable name as requested. See dec6d20